py3status/modules/whatismyip.py

```diff
@@ -61,43 +74,38 @@ def on_click(self, event):
         else:
             self.mode = 'ip'
 
-    def _get_my_ip_and_location(self):
+    def _get_my_ip_info(self):
         """
         """
         try:
-            if self.py3.format_contains(self.format, 'country'):
-                resp = urlopen(self.url_geo, timeout=self.timeout).read()
-                resp = json.loads(resp)
-                country = resp['country']
-                ip = resp['query']
-            else:
-                country = None
-                ip = urlopen(self.url, timeout=self.timeout).read()
-                ip = ip.decode('utf-8')
+            resp = self.py3.request(self.url_geo, timeout=self.timeout).json()
+            info = {}
+            for placeholder in self.py3.get_placeholders_list(self.format):
+                if placeholder in resp:
+                    info[placeholder] = resp[placeholder]
+            for old, new in self.substitutions.items():
+                info[old] = resp.get(new)
+            return info
         except Exception:
-            country = None
-            ip = None
-        return country, ip
+            return None
 
     def whatismyip(self):
         """
         """
-        country, ip = self._get_my_ip_and_location()
+        info = self._get_my_ip_info()
         response = {
             'cached_until': self.py3.time_in(self.negative_cache_timeout)
         }
 
-        if ip is None and self.hide_when_offline:
+        if info is None and self.hide_when_offline:
             response['full_text'] = ''
-        elif ip is not None:
+        elif info is not None:
             response['cached_until'] = self.py3.time_in(self.cache_timeout)
             if self.mode == 'ip':
-                response['full_text'] = self.py3.safe_format(self.format, {
-                    'country': country,
-                    'ip': ip})
+                response['full_text'] = self.py3.safe_format(self.format, info)
             else:
                 response['full_text'] = self.format_online
-                response['color'] = self.py3.COLOR_GOOD
+            response['color'] = self.py3.COLOR_GOOD
```

Comment:
I think this should be moved to line 104 because I feel if we keep it here, someone eventually will view this as a mis-indentation in the future.